dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -2367,124 +2603,249 @@ public void deleteVersion(Contentlet contentlet, User user,boolean respectFronte
 
     @WrapInTransaction
     @Override
-    public void archive(final Contentlet contentlet, final User user, final boolean respectFrontendRoles) throws DotDataException,DotSecurityException, DotContentletStateException {
+    public void archive(final Contentlet contentlet, final User user, final boolean respectFrontendRoles)
+            throws DotDataException, DotSecurityException, DotContentletStateException {
+
         archive(contentlet, user, respectFrontendRoles, false);
     }
 
-    private void archive(final Contentlet contentlet, final User user, final boolean respectFrontendRoles, final boolean isDestroy) throws DotDataException,DotSecurityException, DotContentletStateException {
-        logContentletActivity(contentlet, "Archiving Content", user);
-        try {
+    private Optional<Contentlet> checkAndRunArchiveAsWorkflow(final Contentlet contentletIn, final User user,
```

Comment:
Logic similar to the method `checkAndRunUnpublishAsWorkflow ()`. This can be refactored to use a common method